common/table.js

```diff
@@ -792,17 +812,28 @@
             };
 
             // function for removing a single pill and it's corresponding selected row
-            scope.removePill = function(key) {
+            scope.removePill = function(key, $event) {
                 var index = scope.vm.selectedRows.findIndex(function (obj) {
                     return obj.uniqueId == key;
                 });
-                scope.vm.selectedRows.splice(index, 1);
+
+                // this sanity check is not necessary since we're always calling
+                // this function with a valid key. but it doesn't harm to check
+                if (index === -1) {
+                    $event.preventDefault();
+                    return;
+                }
+
+                var tuple = scope.vm.selectedRows.splice(index, 1)[0];
+                _callOnRowClick(scope, tuple, false);
             };
 
             // function for removing all pills regardless of what page they are on, clears the whole selectedRows array
-            scope.removeAllPills = function() {
+            scope.removeAllPills = function($event) {
```

Comment:
not sure what the point is of passing `$event` here. Looks to be unnecessary unless it's a common angular idiom that we should have been following before..

Reply:
At first, I was using the value that  `_callOnRowClick` is returning. If you look at the `modalDataChanged` which is eventually the function that is being called for clicking on row, it's returning `false` when we hit to the URL limit. In that case, I was actually calling `$event.preventDefault()` which would prevent the action that we were doing.

But later we decided to not call the `preventDefault`. I could remove the `$even` object, but later we might actually want to change this to do the `preventDefault` and I feel like there's no harm in passing the value.